doc/interpolators/cubic_b_spline.qbk

```diff
@@ -0,0 +1,53 @@
+[/
+Copyright (c) 2017 Nick Thompson
+Use, modification and distribution are subject to the
+Boost Software License, Version 1.0. (See accompanying file
+LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+]
+
+[section:cubic_b Cubic B-spline interpolation]
+
+[section:cubic_b_over Overview of Cubic B-Spline Interpolation]
+
+The cubic b spline routine provided by boost allows fast and accurate interpolation of a function which is known at equally spaced points.
+The cubic b-spline interpolation is preferable to traditional cubic spline interpolation as the global support of cubic polynomials makes the interpolation ill-conditioned.
+
+There are many use cases for interpolating a function at equally spaced points.
+One particularly important example is solving ODE's whose coefficients depend on data determined from experiment or numerically.
+Since most ODE steppers are adaptive, they must be able to sample the coefficients at arbitrary points; not just at the points we know the values of our function.
+
+The routine must be provided the following arguments:
```

Comment:
Docs fixed. Overloading operator() is a big improvement to usability and readability.

I didn't implement PIMPL, though I don't object to that. Copying the object is definitely expensive, but is there a use case? Sharing the function pointer to the overloaded () or the `prime` function seems more reasonable.